app/controllers/admin/configs_controller.rb

```diff
@@ -223,6 +221,23 @@ def toggle_email_password_authentication
       end
     end
 
+    def update_enabled_auth_providers(value)
```

Comment:
These private functions exist because the text-field `value` of `auth_providers` received from the front end is a string.

These functions convert this string to an array, checks the validity of each array item, then builds the final array with which `SiteConfig.authentication_providers` is updated.